Components/VxWatchDogTimer/VxWatchDogTimer.cpp

```diff
@@ -0,0 +1,71 @@
+// ======================================================================
+// \title  VxWatchDogTimer.cpp
+// \author ortega
+// \brief  cpp file for VxWatchDogTimer component implementation class
+// ======================================================================
+
+#include "Components/VxWatchDogTimer/VxWatchDogTimer.hpp"
+#include <sysLib.h>
+
+namespace Components {
+
+// ----------------------------------------------------------------------
+// Component construction and destruction
+// ----------------------------------------------------------------------
+
+VxWatchDogTimer ::VxWatchDogTimer(const char* const compName) : VxWatchDogTimerComponentBase(compName) {
+    this->m_watchdogId = wdCreate();
+}
+
+VxWatchDogTimer ::~VxWatchDogTimer() {
+    if (this->m_watchdogId != nullptr) {
+        (void)wdCancel(this->m_watchdogId);
+        (void)wdDelete(this->m_watchdogId);
+    }
+}
+
+void VxWatchDogTimer::startWatchdog(U32 ticks) {
+    this->m_tickDelay = ticks;
+    FW_ASSERT(this->m_watchdogId != nullptr);
+    STATUS status = wdStart(this->m_watchdogId, ticks, reinterpret_cast<FUNCPTR>(s_driverISR),
+                            reinterpret_cast<_Vx_usr_arg_t>(this));
+    FW_ASSERT(status == VXWORKS_OK);
+}
+
+void VxWatchDogTimer::startWatchdog(Fw::TimeInterval interval) {
+    static constexpr U32 USECS_PER_SECS = 1000000;
+    U32 delayInUsecs = (interval.getSeconds() * USECS_PER_SECS) + interval.getUSeconds();
+    // Calculate ticks per interval by multiplying interval by number of ticks per second, then round up.
+    static_assert(USECS_PER_SECS != 0, "This constant cannot be 0.");
```

Comment:
I don't think you need this assert. USECS_PER_SECS must exactly equal 1000000.  This is set by the constant.